lib/ansible/modules/packaging/os/yum_repository.py

```diff
@@ -635,7 +638,7 @@ def main():
         argument_spec=dict(
             async=dict(type='bool'),
             bandwidth=dict(),
-            baseurl=dict(),
+            baseurl=dict(type='raw'),
```

Comment:
Replied by phone so answer ended up in main thread: ```type="list"``` still works because it converts a string into a one element list.The only thing to watch out for with list type is if the string commonly has commas in it.  (list will turn a comma separated string into a list split on commas.  If commas are uncommon but possible [all filesystem paths are like that], the user just needs to make it a list themselves as a workaround:
```
- yum_repository:
    baseurl: ['/srv/rhel,centos-6,7/rpms']
```
)